src/boot/efi/stub.c

```diff
@@ -357,14 +356,69 @@ static void install_addon_devicetrees(
         }
 }
 
+static inline void iovec_array_append(struct iovec **arr, size_t *n_arr, struct iovec elem) {
```

Comment:
Add `assert(arr)` and `assert(n_arr)` here.

Reply:
Maybe name this `iovec_array_extend()` since we have `strv_extend()` as well to do the same for strv.